src/modules/cpu/kernel/jitter.hpp

```diff
@@ -0,0 +1,757 @@
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32u kernelSize = kernelSizeTensor[batchCount];
+        Rpp32u bound = (kernelSize - 1) / 2;
+        Rpp32u heightLimit = roi.xywhROI.roiHeight - bound;
+        Rpp32u offset = batchCount * srcDescPtr->strides.nStride;
+
+        Rpp8u *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp8u *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        Rpp32u alignedLength = roi.xywhROI.roiWidth & ~3;   // Align dst width to process 4 dst pixels per iteration
+        Rpp32u vectorIncrement = 12;
+        Rpp32u vectorIncrementPerChannel = 4;
+        RpptXorwowStateBoxMuller xorwowState;
+        Rpp32s srcLocArray[4] = {0};
+
+        __m128i pxXorwowStateX[5], pxXorwowStateCounter;
+        rpp_host_rng_xorwow_state_offsetted_sse(xorwowInitialStatePtr, xorwowState, offset, pxXorwowStateX, &pxXorwowStateCounter);
+        __m128 pKernelSize = _mm_set1_ps(kernelSize);
+        __m128 pChannel = _mm_set1_ps(layoutParams.bufferMultiplier);
+        __m128 pHStride = _mm_set1_ps(srcDescPtr->strides.hStride);
+        __m128 pHeightLimit = _mm_set1_ps(heightLimit);
+        __m128 pWidthLimit = _mm_set1_ps(roi.xywhROI.roiWidth - 1);
+        __m128 pBound = _mm_set1_ps(bound);
+
+        // Jitter with fused output-layout toggle (NHWC -> NCHW)
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp8u *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+
+            for(int dstLocRow = 0; dstLocRow < roi.xywhROI.roiHeight; dstLocRow++)
+            {
+                Rpp8u *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+                
+                __m128 pRow = _mm_set1_ps(dstLocRow);
+                __m128 pCol = xmm_pDstLocInit;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+                    __m128i pxRow;
+                    compute_jitter_src_loc_sse(pxXorwowStateX, &pxXorwowStateCounter, pRow, pCol, pKernelSize, pBound, pHeightLimit, pWidthLimit, pHStride, pChannel, srcLocArray);
+                    rpp_simd_load(rpp_resize_nn_load_u8pkd3, srcPtrChannel, srcLocArray, pxRow);
```

Comment:
I think the reason why code was using SSE is because the nn interpolation helpers are using SSE loads only
if we need to use helpers for AVX, we need to add corresponding nn interpolation load functions in AVX

@r-abishek Please share your thoughts on if we need to add the load functions in AVX and port the whole code to AVX

Reply:
@r-abishek  please share your thoughts on porting the code to AVX